release notes/v0.37.0.md

```diff
@@ -0,0 +1,61 @@
+}
+```
+
+## Enhancements and UX improvements
+
+- We continued work on the source map feature: add samples ([#2339](https://github.com/grafana/k6/pull/2339)) and improve loading and parsing ([#2355](https://github.com/grafana/k6/pull/2355)).
+- Updated installation from source instructions ([#2359](https://github.com/grafana/k6/pull/2359)). Thanks, @[karitham](https://github.com/Karitham), for making the change.
+- We migrated our docker repository to the [grafana/k6](https://hub.docker.com/r/grafana/k6) ([#2377](https://github.com/grafana/k6/pull/2377)).
+- Introduced a Go-based thresholds parsing ([#2400](https://github.com/grafana/k6/pull/2400)).
+- Enabled more TC39 tests: [optional-chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) ([#2338](https://github.com/grafana/k6/pull/2338)) and async ([#2396](https://github.com/grafana/k6/pull/2396)).
+
+## Bugs fixed!
+
+- Prevent panic in [check](https://k6.io/docs/javascript-api/k6/check-val-sets-tags) if a second argument isn't provided ([#2387](https://github.com/grafana/k6/pull/2387)).
+- `http.head` stopped taking a body as a second argument ([#2402](https://github.com/grafana/k6/pull/2402)). A signature looks like how it was before v0.36.0 `http.head(url, [params])`.
+- Fixed [options.scenarios's](https://k6.io/docs/using-k6/options/#scenarios) JSON marshaling ([#2392](https://github.com/grafana/k6/pull/2392)).
+- Prevent panic in [http.batch](https://k6.io/docs/javascript-api/k6-http/batch-requests) if pass no arguments (https://github.com/grafana/k6/pull/2415).
+
+## Internals
+
+- We updated our CI to make a developer experience better. Dependency and linter checks run only for pull requests ([#2403](https://github.com/grafana/k6/pull/2403)). We also migrated to a native `golangci-lint` GitHub action. Also bumped the windows version ([#2388](https://github.com/grafana/k6/pull/2388)).
+- Updated goja dependency ([#2350](https://github.com/grafana/k6/pull/2350)).
+- The k6's GRPC module was refactored in order to use the new k6's modules API ([#2365](https://github.com/grafana/k6/pull/2365)) and marked headers parameter as deprecated ([#2370](https://github.com/grafana/k6/pull/2370)).
+- This release also contains a bunch of the refactoring PRs that fix linters errors ([#2334](https://github.com/grafana/k6/pull/2334), [#2331](https://github.com/grafana/k6/pull/2331) and [#2341](https://github.com/grafana/k6/pull/2341)), remove global variables' usage ([#2336](https://github.com/grafana/k6/pull/2336), [#2358](https://github.com/grafana/k6/pull/2358), [#2353](https://github.com/grafana/k6/pull/2353) and [#2357](https://github.com/grafana/k6/pull/2357)) and remove unnecessary dependency ([#2313](https://github.com/grafana/k6/pull/2313)) which makes our codebase more consistent and maintainable.
+- Switched `envconfig` to a fork ([#2337](https://github.com/grafana/k6/pull/2337)) in order to abstract OS package and improve testability.
```

Comment:
@na-- you marked the #2337 with a breaking change, but I'm not sure if I got why, could you please help me and clarify it :thinking:

Reply:
Hmm not sure why I did that :confused: Maybe because the original `envconfig` would slurp up unprefixed variables as well, but that shouldn't be a problem anymore since we stopped using prefix in k6? :confused: If @MStoykov doesn't think there are breaking changes in that PR, feel free to remove the label

Reply:
I specifically wanted to not change anything -so there should be no breaking changes AFAIK.

 Looking at the envconfig changes (in the vendor directory) it only includes my changes (which are backwards compatible) and some formatting fixes.